From fcde6480f1b8c9b602b2e571591e73c57c8cc422 Mon Sep 17 00:00:00 2001
From: Piyush Mehta <piyush.mehta@xilinx.com>
Date: Tue, 1 Sep 2020 02:16:20 +0530
Subject: [PATCH 1545/1851] usb: phy: Add NULL checks for return type function

commit d8118dcdd4cc3fc32a8d05dc7f07cfaf386b0a17 from
https://github.com/Xilinx/linux-xlnx.git

To fix the Coverity warning, added NULL value checks for return type
function.

Addresses-Coverity: "Dereference null return value (NULL_RETURNS)"
Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/phy/phy-ulpi.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/phy/phy-ulpi.c b/drivers/usb/phy/phy-ulpi.c
index 0f7f6eb16041..1872c73d67f6 100644
--- a/drivers/usb/phy/phy-ulpi.c
+++ b/drivers/usb/phy/phy-ulpi.c
@@ -316,11 +316,23 @@ static int ulpi_phy_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+	if (!res) {
+		dev_err(&pdev->dev, "no phy I/O memory resource defined\n");
+		return -ENODEV;
+	}
+
 	uphy->regs = devm_ioremap(&pdev->dev, res->start, resource_size(res));
+	if (!uphy->regs) {
+		dev_err(&pdev->dev, "failed to map phy I/O memory\n");
+		return -EFAULT;
+	}
+
 	if (IS_ERR(uphy->regs))
 		return PTR_ERR(uphy->regs);
 
-	ret = of_property_read_u32(np, "view-port", &uphy->vp_offset);
+	if (of_property_read_u32(np, "view-port", &uphy->vp_offset))
+		dev_dbg(&pdev->dev, "Missing view-port property\n");
+
 	if (IS_ERR(uphy->regs)) {
 		dev_err(&pdev->dev, "view-port register not specified\n");
 		return PTR_ERR(uphy->regs);
-- 
2.31.1

